src/types/index.ts

```diff
@@ -12,7 +12,7 @@ export { L1Signer };
 
 export interface L2Signer {
   signMessage(message: string): Promise<string>;
-  getAddress(): string;
+  getAddress(): string | Promise<string>;
```

Comment:
shouldn;t it always be promise?

Reply:
@yundifu 
Refers to the BaseSigner class; the getAddress doesn't have to be promised because it is not async. makes Promise<string> only for the future purpose.

Reply:
@yundifu if you want more information
refer to https://immutable.atlassian.net/browse/WT-614

Reply:
We could always return a Promise of something, even when we don't need it, but I would honestly prefer don't have this workaround.

Reply:
`string | promise<string>` will that be equavient to `promise<string>` when client use?

Reply:
yep, in practice will